java/java-impl/src/com/intellij/codeInsight/FunctionalInterfaceSuggester.java

```diff
@@ -260,14 +251,16 @@ else if (expression instanceof PsiMethodReferenceExpression) {
 
                 for (int i = 0; i < targetMethodParameters.length; i++) {
                   left[i + offset] = parameters[i + offset].getType();
-                  right[i + offset] = partialSubstitutor.substitute(targetMethodParameters[i].getType());
```

Comment:
This substitution is handled inside of `com.intellij.psi.PsiResolveHelper#inferTypeArguments()`